push/sched/scheduler.go

```diff
@@ -2,29 +2,41 @@ package sched
 
 import (
 	log "github.com/sirupsen/logrus"
-	"sync"
+	"github.com/squareup/pranadb/cluster"
+	"github.com/squareup/pranadb/common"
+	"sync/atomic"
+	"time"
 )
 
+const maxProcessBatchRows = 500
+
 type ShardScheduler struct {
-	shardID uint64
-	actions chan *actionHolder
-	lock    sync.Mutex
-	started bool
-	paused  bool
+	shardID                  uint64
+	actions                  chan struct{}
+	started                  bool
+	failed                   bool
+	lock                     common.SpinLock
```

Comment:
It's not clear why we switched from the mutex to a spin-lock. Do you mind adding a comment for future reference?